### PR TITLE
FIX: Trino&Presto catalogs as databases

### PR DESCRIPTION
**Pull request by MilanBariya**
### Describe your changes :

I worked on the fixing Trino&Presto catalogs as databases.
Fix: #8158

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/database/trino.py [hunk 1]
```diff
@@ -154,6 +160,48 @@ def create(cls, config_dict, metadata_config: OpenMetadataConnection):
...
+
+        new_service_connection = deepcopy(self.service_connection)
+        new_service_connection.catalog = database_name
+        self.engine = get_connection(new_service_connection)
         self.inspector = inspect(self.engine)
-        yield self.trino_connection.catalog
+
+    def get_database_names(self) -> Iterable[str]:
+        configured_catlog = self.trino_connection.catalog
+        if configured_catlog:
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/database/trino.py:178]: nit - typo in `catlog`

Comment (pmbrull) [reply ingestion/src/metadata/ingestion/source/database/trino.py:178]: needs to be changed in both trino and presto

Comment (MilanBariya) [reply ingestion/src/metadata/ingestion/source/database/trino.py]: Solved in both.🙏

**Review by TeddyCr — APPROVED** [review]

**Comment by TeddyCr** [comment]
@MilanBariya have we checked that connecting without a catalog was ok for older versions of Trino/Presto?

**Comment by MilanBariya** [comment]
> @MilanBariya have we checked that connecting without a catalog was ok for older versions of Trino/Presto?

@TeddyCr I test this using the latest docker image of both.

**Comment by MilanBariya** [comment]
> @MilanBariya have we checked that connecting without a catalog was ok for older versions of Trino/Presto?

@TeddyCr I test out with an older version of both and it's working without catalog also.

**Pull request merged**